src/test/java/com/techsophy/tsf/account/controller/InternalControllerImplTest.java

```diff
@@ -0,0 +1,111 @@
+package com.techsophy.tsf.account.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.techsophy.idgenerator.IdGeneratorImpl;
+import com.techsophy.tsf.account.config.GlobalMessageSource;
+import com.techsophy.tsf.account.controller.impl.InternalControllerImpl;
+import com.techsophy.tsf.account.dto.UserData;
+import com.techsophy.tsf.account.dto.UserFormDataSchema;
+import com.techsophy.tsf.account.entity.UserFormDataDefinition;
+import com.techsophy.tsf.account.repository.UserDefinitionRepository;
+import com.techsophy.tsf.account.repository.UserFormDataDefinitionRepository;
+import com.techsophy.tsf.account.service.impl.UserFormDataServiceImpl;
+import com.techsophy.tsf.account.service.impl.UserServiceImpl;
+import com.techsophy.tsf.account.utils.Rsa4096;
+import com.techsophy.tsf.account.utils.TokenUtils;
+import com.techsophy.tsf.account.utils.UserDetails;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.math.BigInteger;
+import java.security.Signature;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.techsophy.tsf.account.constants.AccountConstants.*;
+import static com.techsophy.tsf.account.constants.PropertyConstant.*;
+
+@ExtendWith({MockitoExtension.class, SpringExtension.class})
+class InternalControllerImplTest {
```

Comment:
Please add test cases when
- signature is not present
- signature is not verified
- signature is verified
- if user service throws an exception
- etc

Reply:
If key is not found
key is invalid